test/numa/tests/memtier/n4c16/test01-only-add-pods/code.var.sh

```diff
@@ -0,0 +1,125 @@
+# Test that guaranteed and burstable pods get the CPUs they require
+# when there are enough CPUs available.
+
+# pod0, fits in a core
+CPU=1 create guaranteed
+report allowed
+verify \
+    'len(cpus["pod0c0"]) == 1'
+
+# pod1, takes full core
+CPU=2 create guaranteed
+report allowed
+verify \
+    'len(cpus["pod0c0"]) == 1' \
+    'len(cpus["pod1c0"]) == 2' \
+    'len(cores["pod1c0"]) == 1' \
+    'len(set.union(cpus["pod0c0"], cpus["pod1c0"])) == 3'
```

Comment:
For the simplest basic cases, shouldn't we spell out/check the explicit ID of the cpu we expect to get allocated ? It is pretty well defined and intentional behavior, after all.

Reply:
Ok. I've tried to avoid failing a test when pinning is any of the optimal ones. But sure, if you'd prefer fixing CPU ids, then I'll change the most obvious cases to that.

Reply:
Well, there is a deliberate attempt to write the scoring algorithm so that it never evaluates the score of two alternatives as equal. Everything else being equal, the CPU with the lower id wins. This was mostly put in with testing in mind, as it makes allocation runs reproducible with predictable results. Since this is deliberate behavior, I think we should have at least some cases verifying it.

Reply:
We can also handle this with further PRs after this one is merged. It's been pending long enough and start to be really good for testing.

Reply:
Ok, let's take further CPU id verifications later, I added a couple now.